web/components/controls/time-picker.jsx

```diff
@@ -31,84 +38,142 @@ class TimePicker extends React.Component {
 		this.onValueChanged = this.onValueChanged.bind(this);
 	}
 
-	onValueChanged(e) {
-		const time = moment(this.props.getValue());
-		const isPm = (time.format('a') === 'pm');
-		let hour = time.hour();
-		let minute = time.minute();
+	componentWillReceiveProps(nextProps) {
+		const value = nextProps.getValue();
+
+		if (value) {
+			const time = moment(value);
+
+			this.setState({
+				hour: time.format('h'),
+				minute: time.format('m'),
+				ampm: time.format('a')
+			});
+		}
+	}
 
+	onValueChanged(e) {
+		const currentState = Object.assign({}, this.state);
 		switch (e.target.id) {
 			case this.props.controlId + '_hours':
-				hour = Number.parseInt(e.target.value) % 12;
-				if (isPm) hour += 12;
+				currentState.hour = e.target.value;
 				break;
 
 			case this.props.controlId + '_minutes':
-				minute = Number.parseInt(e.target.value);
+				currentState.minute = e.target.value;
 				break;
 
 			case this.props.controlId + '_ampm':
-				if (e.target.value === 'am') hour -= 12;
-				else hour += 12;
+				currentState.ampm = e.target.value;
 				break;
 		}
 
-		const value = time.hour(hour).minute(minute).toISOString();
+		this.setState(currentState);
 
-		if (!this.props.onChange) {
-			this.props.setValue(value);
-		} else {
-			this.props.onChange(value);
+		if (!currentState.hour || !currentState.minute) {
+			// Don't show the "Required Field" error until a value has been set or the form
+			// has been submitted.
+			if (this.props.isPristine()) return;
+
+			return this.props.setValue(null);
+		}
+
+		let time = moment()
```

Comment:
Seconds will be included in here... but we probably don't care. Give it some thought.